format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java

```diff
@@ -79,11 +79,21 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     // invalid
     collector.getOrThrowException();
 
+    if (config.containsMacro(NAME_FORMAT)) {
+      // Deploy all format plugins. This ensures that the required plugin is available when
+      // the format macro is evaluated in prepareRun,
+      for (FileFormat f: FileFormat.values()) {
+        pipelineConfigurer.usePlugin(ValidatingInputFormat.PLUGIN_TYPE, f.name().toLowerCase(), f.name().toLowerCase(),
+          config.getRawProperties());
+      }
```

Comment:
we should have similar error handling here in the source if there is an error at configure or prepare time.

Reply:
fixed